src/dcp/dcp.c

```diff
@@ -267,14 +415,99 @@ int main(int argc, char** argv)
     if (mfu_progress_timeout < 0) {
         if (rank == 0) {
             MFU_LOG(MFU_LOG_ERR, "Seconds in --progress must be non-negative: %d invalid", mfu_progress_timeout);
+            usage = 1;
         }
-        usage = 1;
```

Comment:
Need to keep usage=1 here, or otherwise we'll deadlock.  All ranks need to know.